lib/device-connections-factory.js

```diff
@@ -23,11 +23,14 @@ class iProxy {
       return;
     }
 
-    const remoteSocketPromise = utilities.connectPort(this.udid, this.deviceport)
-      .catch((e) => void this.log.debug(e.message));
     this.localServer = net.createServer(async (localSocket) => {
-      const remoteSocket = await remoteSocketPromise;
-      if (!remoteSocket) {
+      let remoteSocket;
+      try {
+        // 'Connection was refused to port 8100' error loop never finish
```

Comment:
I'd rather say:
We can only connect to the remote socket after the local socket connection succeeds